app/ui-utils/client/lib/RoomHistoryManager.js

```diff
@@ -235,6 +235,7 @@ export const RoomHistoryManager = new (class extends Emitter {
 		waitAfterFlush(() => {
 			const heightDiff = wrapper.scrollHeight - previousHeight;
 			wrapper.scrollTop = scroll + heightDiff;
+			lastMessage && document.getElementById(lastMessage._id).scrollIntoView({ behavior: 'smooth', block: 'end', inline: 'nearest' });
```

Comment:
This scrolls the dom twice, which is not very efficient 🤔

Reply:
Thanks a lot for finally reviewing this PR. Actually, in both (before and after changes) cases DOM scrolls twice, the difference is in here it scrolls smoothly.

Reply:
When the message box is refreshed with history the whole view gets to the top, so to *fix* that the view is then scrolled back down with `Element.scrollTop`. So yes it **was** scrolling twice already, but now it's doing it three times - 1) scrolling to the top with all the new messages 2) scrolling down to get the user to the expected position 3) scrolling again because the last one didn't quite work as expected - I just didn't count the initial scroll the first time.

does this make sense?

Reply:
Yeah, now I understand. So, now I have changed the current logic to adapt to the currently used code structure. Now, after the new messages are loaded the DOM is scrolled back such that the last visible message before loading new messages appears at the top, here is a video, showing the behavior.

https://user-images.githubusercontent.com/61188295/156872481-74addaa1-0038-45d8-874e-948164789d84.mp4

Please review once again. Thank you!